### PR TITLE
Improved '\dump_state' command.

### PR DESCRIPTION
**Pull request by kukabu**
Hello

@oddstr13, can you check it? I think fldigi must works now.

### CODE REVIEW

**Comment by kukabu** [comment]
@csete, I want to refactoring  the RemoteControl::startRead(), one command - one function. Now it's a very large function and unclear. What is your opinion?

**Comment by kukabu** [comment]
I tested gqrx+fldigi via pulseaudio. Looks like it works as expected with one comment, hamlib backend sends 'v' for get current VFO. It's easy to fix.

**Comment by csete** [comment]
@kukabu,

I thought we agreed that you will provide more information with your patches so that I (and others) can evaluate them without having to dig into the source code. "Improved xyz..." is just not good enough and leaves it to me to guess why this is an improvement. As we become more people who have to work on the code repository, we have to be able to understand the commits when we look at it later.

You are welcome to study this brief article about writing commit messages:
http://chris.beams.io/posts/git-commit/

In this case it would seem to be very relevant to include information about compatibility with fldigi, for example:

> Improve '\dump_state' command
>
> These changes are necessary to make fldigi work with the remote interface.

With a commit message like this I know exactly why the change is necessary and so will everybody else who looks at it later.

Sorry if I sound like an idiot. I am very busy at the moment and have basically no time to work on gqrx at all. I planned to take a break from gqrx for the rest of the year, so, if I am to keep spending time on reviewing patches now it has to be much easier for me to review them. Otherwise they will just have to wait until I have better time.

**Comment by csete** [comment]
> I want to refactoring the RemoteControl::startRead(), one command - one function. Now it's a very large function and unclear. What is your opinion?

Good idea. You can start with '\dump_state' ;-)

File: src/applications/gqrx/remote_control.cpp [hunk 1]
```diff
@@ -366,26 +366,73 @@ void RemoteControl::startRead()
...
+                          * start, end, modes, low_power, high_power, vfo, ant
+                          *  start/end - Start/End frequency [Hz]
+                          *  modes - Bit field of RIG_MODE's (AM|CW|CWR|USB|LSB|FM|WFM)
+                          *  low_power/high_power - Lower/Higher RF power in mW,
+                          *                         -1 for no power (ie. rx list)
+                          *  vfo - VFO list equipped with this range (RIG_VFO_MAIN)
+                          *  ant - Antenna list equipped with this range, 0 means all
+                          *  FIXME: limits can be gets from receiver::get_rf_range()
+                          */
+                         "1.000000 1000000000.000000 0xef -1 -1 0x4000000 0x0\n"
```
Comment (oddstr13) [inline src/applications/gqrx/remote_control.cpp]: HackRF goes up to around 6 GHz

**Comment by oddstr13** [comment]
@kukabu I'll take it for a test run when I boot into Linux next time :)

**Comment by oddstr13** [comment]
See also issue #329

File: src/applications/gqrx/remote_control.cpp (hunk 1 above)
Comment (csete) [inline src/applications/gqrx/remote_control.cpp]: I am aware of these issues but currently we do not have the framework to use the correct parameters depending on which device is being used. We can add it later if it is really absolutely necessary, but I want to keep this practical and only add thins that are necessary to make applications work.

Comment (kukabu) [inline src/applications/gqrx/remote_control.cpp]: ups I lost zero, should be 10 for upper limit and I think we can use 0 for lower limit.

**Comment by kukabu** [comment]
@csete Thanks for your comments, I will take into account your requirements.

**Comment by kukabu** [comment]
I change vfo field from RIG_VFO_MAIN to RIG_VFO_A, it's more correct definition.

**Comment by csete** [comment]
Ok, looks much better now, thanks :-)

I don't think we can use 10 GHz because the limit on number of digits c.f. issue #343 , but I intend to fix it for 2.7 anyway.

**Comment by oddstr13** [comment]
Having tested this some now, I conclude that fldigi 3.23.15 is very much not well behaved when it comes to the Hamlib NET rigctl.
It seems to be sending 1 character per packet.. sometimes.
```hex
00000000  5c                                                 \
00000001  64                                                 d
00000002  75                                                 u
    00000000  52 50 52 54 20 31 0a                               RPRT 1.
00000003  6d                                                 m
00000004  70 5f 73 74 61 74 65 0a                            p_state. 
    00000007  52 50 52 54 20 31 0a                               RPRT 1.
0000000C  71                                                 q
0000000D  0a                                                 .
```
```hexdump
00000000  5c                                                 \
00000001  64                                                 d
00000002  75                                                 u
00000003  6d                                                 m
00000004  70                                                 p
00000005  5f                                                 _
00000006  73                                                 s
00000007  74                                                 t
00000008  61                                                 a
00000009  74                                                 t
0000000A  65                                                 e
0000000B  0a                                                 .
    00000000  52 50 52 54 20 31 0a                               RPRT 1.
0000000C  71                                                 q
0000000D  0a                                                 .
```

Compare this with a netcat connection;
```
00000000  5c 64 75 6d 70 5f 73 74  61 74 65 0a               \dump_st ate.
    00000000  30 0a 32 0a 31 0a 30 2e  30 30 30 30 30 30 20 31   0.2.1.0. 000000 1
    00000010  30 30 30 30 30 30 30 30  30 30 2e 30 30 30 30 30   00000000 00.00000
    00000020  30 20 30 78 65 66 20 2d  31 20 2d 31 20 30 78 31   0 0xef - 1 -1 0x1
    00000030  20 30 78 30 0a 30 20 30  20 30 20 30 20 30 20 30    0x0.0 0  0 0 0 0
    00000040  20 30 0a 30 20 30 20 30  20 30 20 30 20 30 20 30    0.0 0 0  0 0 0 0
    00000050  0a 30 78 65 66 20 31 0a  30 78 65 66 20 30 0a 30   .0xef 1. 0xef 0.0
    00000060  20 30 0a 30 78 38 32 20  35 30 30 0a 30 78 38 32    0.0x82  500.0x82
    00000070  20 32 30 30 0a 30 78 38  32 20 32 30 30 30 0a 30    200.0x8 2 2000.0
    00000080  78 32 31 20 31 30 30 30  30 0a 30 78 32 31 20 35   x21 1000 0.0x21 5
    00000090  30 30 30 0a 30 78 32 31  20 32 30 30 30 30 0a 30   000.0x21  20000.0
    000000A0  78 30 63 20 32 37 30 30  0a 30 78 30 63 20 31 34   x0c 2700 .0x0c 14
    000000B0  30 30 0a 30 78 30 63 20  33 39 30 30 0a 30 78 34   00.0x0c  3900.0x4
    000000C0  30 20 31 36 30 30 30 30  0a 30 78 34 30 20 31 32   0 160000 .0x40 12
    000000D0  30 30 30 30 0a 30 78 34  30 20 32 30 30 30 30 30   0000.0x4 0 200000
    000000E0  0a 30 20 30 0a 30 0a 30  0a 30 0a 30 0a 30 0a 30   .0 0.0.0 .0.0.0.0
    000000F0  0a 30 0a 30 0a 30 78 34  30 30 30 30 30 32 30 0a   .0.0.0x4 0000020.
    00000100  30 78 32 30 0a 30 0a 30  0a                        0x20.0.0 .
0000000C  66 0a                                              f.
    00000109  39 36 33 39 33 30 30 30  0a                        96393000 .
0000000E  46 39 36 34 39 39 39 39  39 0a                     F9649999 9.
    00000112  52 50 52 54 20 31 0a                               RPRT 1.
00000018  46 20 39 36 34 30 30 30  30 30 0a                  F 964000 00.
    00000119  52 50 52 54 20 30 0a                               RPRT 0.
00000023  71 0a                                              q.
```

I'm not sure if this is intended behavior of fldigi, or if it is a bug. Or it might just be a edge case on my particular setup.

**Comment by kukabu** [comment]
@csete, Thank!
@oddstr13, The fldigi uses a very small delays, try this
```
I: setConf: setting "post_write_delay" to "500"
I: setConf: setting "write_delay" to "0"
I: setConf: setting "timeout" to "1000"
I: setConf: setting "retry" to "2"
```
I think TCP_NODELAY  should help but this requires further investigation.

**Comment by csete** [comment]
If there are still problems with fldigi we should perhaps open a new bug and record the findings there.

**Pull request merged**